src/main/java/io/fusionauth/samlv2/service/DefaultSAMLv2Service.java

```diff
@@ -753,7 +753,28 @@ <T> String buildPostRequest(JAXBElement<T> object, Class<T> type, boolean sign,
     if (sign) {
       try {
         Element toSign = document.getDocumentElement();
-        signXML(privateKey, certificate, algorithm, xmlSignatureC14nMethod, toSign, null, includeKeyInfo);
+        Node insertBefore = null;
+        // This method is used for AuthnRequestType, LogoutRequestType, and StatusResponseType (LogoutResponse)
+        // The only required element in the StatusResponseType is Status. See Section 3.2.2 in SAML Core.
+        // The children will be a sequence that must exist in the order of 'Issuer', 'Signature', 'Extensions', and then 'Status'
+        // - If the first element is 'Issuer', then the next sibling will be used for 'insertBefore'.
+        // - If the first element is NOT 'Issuer', it MUST be 'Extensions' or 'Status', and thus is the 'insertBefore' node.
+        //
+        // AuthnRequestType and LogoutRequestType both extend RequestAbstractType.
+        //  - AuthnRequestType has no required fields. See Section 3.4.1 in SAML Core.
+        //  - LogoutRequestType only requires an ID field. See Section 3.7.1 in SAML Core.
+        // The RequestAbstractType requires its children in the sequence of 'Issuer', 'Signature', and then 'Extensions'. See Section 3.2.1 in SAML Core.
+        // - If the first element is 'Issuer', then the next sibling will be used for 'insertBefore'.
+        // - If the first element is NOT 'Issuer', it MUST be 'Extensions', and thus is the 'insertBefore' node.
+        NodeList children = toSign.getChildNodes();
```

Comment:
This uses the same logic as signing the SAML `Response` (as opposed to the `Assertion`).

All of the types that hit this code have an optional `Issuer` which must be followed immediately by the `Signature` (assuming the document is signed).

Reply:
Apart from the comments, this code is identical to the code that starts on line `340`, can we move this to a private method and re-use it? 

If you put your cursor on line `340` and use the IJ intent, it will show you the duplicate, and it should be showing you a yellow indicator in the gutter as well.